internal/host/validator.go

```diff
@@ -1172,8 +1172,9 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
 	}
 }
 
-func isNonePlatformMultiNodeInstallation(c *validationContext) bool {
-	return swag.BoolValue(c.cluster.UserManagedNetworking) && !common.IsSingleNodeCluster(c.cluster)
+func shouldValidateAPIDnsResolution(c *validationContext) bool {
+	networkCfg := network.Config{}
+	return swag.BoolValue(c.cluster.UserManagedNetworking) && !common.IsSingleNodeCluster(c.cluster) && !networkCfg.EnableSingleNodeDnsmasq
```

Comment:
I think I can come up with situations in which this condition doesn't make sense, can you break it up like this:

```go
if !swag.BoolValue(c.cluster.UserManagedNetworking) {
    // No need to perform DNS validations if network is not user managed
    return false;
}

// User managed networking requires DNS validation, unless it's SNO with the dnsmasq configuration
isSnoWithDnsmasq := common.IsSingleNodeCluster(c.cluster) && networkCfg.EnableSingleNodeDnsmasq

return !isSnoWithDnsmasq
```